src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -0,0 +1,176 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.controller;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.UUID;
+
+import codeu.model.data.User;
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+
+import java.util.List;
+import java.util.ArrayList;
+import codeu.model.store.basic.UserStore;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class AdminServletTest {
+
+  private AdminServlet adminServlet;
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private RequestDispatcher mockRequestDispatcher;
+  private UserStore mockUserStore;
+  private ConversationStore mockConvoStore;
+  private MessageStore mockMessageStore;
+
+  @Before
+  public void setup() {
+    adminServlet = new AdminServlet();
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+
+    mockUserStore = Mockito.mock(UserStore.class);
+    mockConvoStore = Mockito.mock(ConversationStore.class);
+    mockMessageStore = Mockito.mock(MessageStore.class);
+
+    User fakeUser =
+        new User(
+            UUID.randomUUID(),
+            "test username",
+            "$2a$10$bBiLUAVmUFK6Iwg5rmpBUOIBW6rIMhU1eKfi3KR60V9UXaYTwPfHy",
+            Instant.now());
+
+    mockUserStore.addUser(fakeUser);
+
+    adminServlet.setUserStore(mockUserStore);
+    adminServlet.setConversationStore(mockConvoStore);
+    adminServlet.setMessageStore(mockMessageStore);
+
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/admin.jsp"))
+        .thenReturn(mockRequestDispatcher);
+  }
+
+  @Test
+  public void testDoGet() throws IOException, ServletException {
+    adminServlet.doGet(mockRequest, mockResponse);
+    mockRequest.getParameter("username");
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testGetStats() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn(null);
+
+    UUID fakeConversationId = UUID.randomUUID();
+    List<Conversation> fakeConversationList = new ArrayList<>();
+    Conversation fakeConversation =
+        new Conversation(fakeConversationId, UUID.randomUUID(), "test_conversation", Instant.now());
+    
+    fakeConversationList.add(fakeConversation);
+
+    Mockito.when(mockConvoStore.getAllConversations())
```

Comment:
Like Gonzalo, I'm still trying to understand Mockito so most of my comments will also be questions. So what are you trying to accomplish with line 96? Is it just saying that when it gets all the conversations it returns the fake conversations?

Reply:
Cool yeah thats exactly what its doing. In order to test this, I had to create a fake conversation store along with a fake conversation with fake messages and fake users. It got really involved real quick.

Reply:
Ah okay, thanks!